### PR TITLE
Submodule treatment for dccrg

### PR DESCRIPTION
**Pull request by alhom**
Continuing the submoduling of header libraries after #865.

Added DCCRG as a submodule, makefiles updated.

### CODE REVIEW

**Comment by markusbattarbee** [comment]
Worth merging dev into this I guess and making the yaml file match how the fsgrid PR ended up

**Review by markusbattarbee — APPROVED** [review]
Looks good!

**Comment by ykempf** [comment]
Well, you asked for my input... I would have appreciated if the Makefiles had a variable for the path to submodules and not the hardcoded `./submodules/` on several lines. But if we don't want to go for that, I can do the click...

File: .gitmodules [hunk 1]
```diff
@@ -3,3 +3,6 @@
 	url = git@github.com:fmihpc/fsgrid.git
 
 
+[submodule "submodules/dccrg"]
```
Comment (ykempf) [inline .gitmodules:6]: So how does this work? I don't see where it specifies that it should land on the `vlasiator-version` branch.

File: submodules/dccrg
Comment (alhom) [inline submodules/dccrg:1]: @ykempf Here - these point to commits primarily.

Comment (ykempf) [reply submodules/dccrg:1]: Well hidden and in particular not highlighted in green.

**Pull request merged**